components/org.wso2.carbon.identity.scim2.common/src/main/java/org/wso2/carbon/identity/scim2/common/impl/SCIMUserManager.java

```diff
@@ -486,90 +486,360 @@ public User updateUser(User user, Map<String, Boolean> requiredAttributes) throw
+                    domainName, filteredUsers);
+        } else {
+            throw new CharonException("Unknown operation. Not either an expression node or an operation node.");
+        }
+    }
+
+    /**
+     * Method to filter users for a filter with a single attribute.
+     *
+     * @param node               Expression node for single attribute filtering
+     * @param requiredAttributes Required attributes for the response
+     * @param offset             Starting index of the count
+     * @param limit              Counting value
+     * @param sortBy             SortBy
+     * @param sortOrder          Sorting order
+     * @param domainName         Domain to run the filter
+     * @return User list with detailed attributes
+     * @throws CharonException Error while filtering
+     */
+    private List<Object> getSingleAttributeFilteredUsers(ExpressionNode node, Map<String, Boolean> requiredAttributes,
+            int offset, int limit, String sortBy, String sortOrder, String domainName) throws CharonException {
+
+        String[] userNames;
+
+        // Set filter values.
+        String attributeName = node.getAttributeValue();
+        String filterOperation = node.getOperation();
+        String attributeValue = node.getValue();
+
+        if (log.isDebugEnabled()) {
+            log.debug(
+                    String.format("Listing users by filter: %s %s %s", attributeName, filterOperation, attributeValue));
+        }
+
+        try {
+
+            validateFilterOperation(filterOperation);
+
+            // Apply filter enhancements for username and update expression node value.
+            attributeValue = applyFilterEnhancementsForUsername(filterOperation, attributeName, attributeValue, node);
+
+            if ((carbonUM instanceof PaginatedUserStoreManager) && isPaginatedUserStoreAvailable()) {
+
+                // Set the limit if specified.
+                if (!(limit > 0))
+                    limit = getMaxLimit();
+
+                // Extract the domain name if the domain name is embedded in the filter attribute value.
+                domainName = resolveDomainInAttributeValueAndUpdateExpressionNode(domainName, node);
+
+                // Get filtered user names.
+                userNames = getFilteredUsers(node, offset, limit, sortBy, sortOrder, domainName);
+            } else
+                userNames = filterUsersUsingLegacyAPIs(filterOperation, attributeName, attributeValue, limit, offset);
+
+        } catch (NotImplementedException e) {
+            String errorMessage = String.format("System does not support filter operator: %s", filterOperation);
+            throw new CharonException(errorMessage, e);
+        } catch (BadRequestException e) {
+            String errorMessage = String
+                    .format("Domain parameter %s in request does not match with the domain name %s in the attribute "
+                            + "value. ", domainName, attributeValue);
```

Comment:
Shall we emphasize the parameter values by adding a colon? You can change `Domain parameter %s` to `Domain parameter: %s`. Same with the others.

Reply:
Changed